src/www/views/servers_view/server_list_item/index.ts

```diff
@@ -0,0 +1,55 @@
+import {LitElement} from 'lit';
+import {property} from 'lit/decorators.js';
+
+import {ServerConnectionState} from '../server_connection_indicator';
+
+export enum ServerListItemEvent {
+  CONNECT = 'ConnectPressed',
+  DISCONNECT = 'DisconnectPressed',
+  FORGET = 'ForgetPressed',
+  RENAME = 'ShowServerRename',
+}
+
+export interface ServerListItem {
+  disabled: boolean;
+  errorMessageId?: string;
+  isOutlineServer: boolean;
+  address: string;
+  id: string;
+  name: string;
+  connectionState: ServerConnectionState;
+}
+
+export class ServerListItemElement extends LitElement {
+  @property() server: ServerListItem;
+
+  get isConnected() {
+    return [
+      ServerConnectionState.CONNECTING,
+      ServerConnectionState.CONNECTED,
+      ServerConnectionState.RECONNECTING,
+    ].includes(this.server.connectionState);
+  }
+
+  dispatchServerRenameEvent() {
+    this.dispatchEvent(
+      new CustomEvent(ServerListItemEvent.RENAME, {detail: this.server, bubbles: true, composed: true})
```

Comment:
in lit you must explicitly say if an event bubbles or is composed, which i like, and should make debugging our D&D issue easier (in theory)